src/gui/application.cpp

```diff
@@ -661,46 +538,122 @@ static void displayHelpText(const QString &t, std::ostream &stream = std::cout)
-}
+    if (parser.isSet(helpOption)) {
+        showHint(parser.helpText());
+    }
 
-void Application::listAvailableTranslations()
-{
-    auto availableTranslations = Translations::listAvailableTranslations().toList();
-    availableTranslations.sort(Qt::CaseInsensitive);
-    displayHelpText("Available translations: " + availableTranslations.join(", "));
+    if (parser.isSet(versionOption)) {
+        displayHelpText(_theme->versionSwitchOutput());
+        ::exit(0);
+    }
+
+    // TODO: rename this option (see #8234 for more information)
+    if (parser.isSet(showSettingsOption)) {
+        _showSettings = true;
+    }
+    if (parser.isSet(quitInstanceOption)) {
+        _quitInstance = true;
+    }
+    if (parser.isSet(logFileOption)) {
+        _logFile = parser.value(logFileOption);
+    }
+    if (parser.isSet(logDirOption)) {
+        _logDir = parser.value(logDirOption);
+    }
+    if (parser.isSet(logExpireOption)) {
+        _logExpire = std::chrono::hours(parser.value(logExpireOption).toInt());
+    }
+    if (parser.isSet(logFlushOption)) {
+        _logFlush = true;
+    }
+    if (parser.isSet(logDebugOption)) {
+        _logDebug = true;
+    }
+    if (parser.isSet(confDirOption)) {
+        const auto confDir = parser.value(confDirOption);
+        if (!ConfigFile::setConfDir(confDir)) {
+            showHint(tr("Invalid path passed to --confdir"));
+        }
+    }
+    if (parser.isSet(debugOption)) {
+        _logDebug = true;
+        _debugMode = true;
+    }
+    if (parser.isSet(languageOption)) {
+        const auto languageValue = parser.value(languageOption);
+
+        // fail if the language is unknown
+        if (!Translations::listAvailableTranslations().contains(languageValue)) {
+            showHint(tr("Error: unknown language \"%1\" (use --list-languages to get a complete list of supported translations)").arg(languageValue));
+        } else {
+            _userEnforcedLanguage = languageValue;
+        }
+    }
+    if (parser.isSet(listLanguagesOption)) {
+        auto availableTranslations = Translations::listAvailableTranslations().toList();
+        availableTranslations.sort(Qt::CaseInsensitive);
+        showHint(tr("Available translations: %1").arg(availableTranslations.join(", ")));
+    }
+
+    auto positionalArguments = parser.positionalArguments();
+
+    // ignore any positional arguments beyond the first one
+    if (!positionalArguments.empty()) {
+        QTimer::singleShot(0, this, [this, positionalArguments] { openVirtualFile(positionalArguments.front()); });
+    }
 }
 
 void Application::showHint(const QString &errorHint)
 {
     QString out;
     QTextStream hint(&out);
     hint << errorHint << endl
-         << "Try '" << QFileInfo(QCoreApplication::applicationFilePath()).fileName() << " --help' for more information" << endl;
+         << tr("Try \"%1\" --help' for more information").arg(QFileInfo(QCoreApplication::applicationFilePath()).fileName()) << endl;
```

Comment:
no need for that stream

Reply:
The latest commit removes the function anyway.